### ISSUE TITLE
alpha and optional combinations

### ISSUE THREAD

**Issue by sajcics** [icebob/fastest-validator#256]
WHAT?
When I define alpha parameter in the optional field, the validator throws error. 

EXAMPLE:
Here is an example how I defined rule for name field. 
`   name: {
      type: 'string',
      optional: true,
      alpha: true,
      trimLeft: true
    },`
    
  the validator throws error `field must be an alphabetic string`. If I remove `alpha` then validation on empty field works.
    
    Expecting:
    When `optional` is set then `alpha` will not validate on empty string.

**Comment by erfanium** [comment]
It's expected, empty string is not a valid `alpha` pattern

**Comment by sajcics** [comment]
yes, but I have an `optional` set to true, which means that alpha should not be applied if string is empty.

**Comment by erfanium** [comment]
`optional` flag only cares about `null` and `undefined` values and doesn't skip empty strings

**Comment by sajcics** [comment]
it will be great if empty string also consider as null and undefined for `optional`...because it's empty and do not representing anything

**Comment by erfanium** [comment]
You can add a preValidation middleware to delete these keys from object. It's better It's more safe for the validator to be strict
